### DISCUSSION TITLE
Leftwm and Eclipse IDE

### DISCUSSION THREAD

**Discussion by Fay3V** [leftwm/leftwm#881] [General]
Hi All,
I want to thank you for the good job done by all of you with leftwm.
It's the first tiling window manager I tried and until now everything work fine for me.
But I face a little issue when I try to run an Eclipse IDE.

I use a  development build of leftwm (commit: [9179969](https://github.com/leftwm/leftwm/commit/917996946d1197beb08acc84115ad2ee6befee7a))

I have installed an eclipse IDE from this link https://ftp.fau.de/eclipse/oomph/epp/2022-09/R/eclipse-inst-jre-linux64.tar.gz
I added a window_rules in my config.ron file in order to spawn it as a floating window.
`(window_class: "Eclipse", spawn_floating: true)`

But the window I get is too small

<img width="252" alt="2022-10-07 17_06_38-wks  Running  - Oracle VM VirtualBox" src="https://user-images.githubusercontent.com/20566921/194587670-96ab8ca6-5788-4c27-a458-f84942773f06.png">

Any further Dialog box opened, like the Search Dialog, is also too small.

this is the result of the xprop command

> _NET_WM_DESKTOP(CARDINAL) = 1
WM_STATE(WM_STATE):
		window state: Normal
		icon window: <field not available>
_NET_WM_STATE(ATOM) =
WM_HINTS(WM_HINTS):
		Client accepts input or input focus: True
		Initial state is Normal State.
		bitmap id # to use for icon: 0x38001fc
		bitmap id # of mask for icon: 0x3800202
		window id # of group leader: 0x3800001
_GTK_THEME_VARIANT(UTF8_STRING) =
_NET_WM_ICON(CARDINAL) =
XdndAware(ATOM) = BITMAP
_NET_WM_OPAQUE_REGION(CARDINAL) = 0, 0, 497, 198
_NET_WM_WINDOW_TYPE(ATOM) = _NET_WM_WINDOW_TYPE_NORMAL
_NET_WM_SYNC_REQUEST_COUNTER(CARDINAL) = 58720762, 58720763
_NET_WM_USER_TIME(CARDINAL) = 11132927
_NET_WM_USER_TIME_WINDOW(WINDOW): window id # 0x38001f9
WM_CLIENT_LEADER(WINDOW): window id # 0x3800001
_NET_WM_PID(CARDINAL) = 9025
WM_LOCALE_NAME(STRING) = "en_US.UTF-8"
WM_CLIENT_MACHINE(STRING) = "archlinux"
WM_NORMAL_HINTS(WM_SIZE_HINTS):
		program specified minimum size: 438 by 27
		program specified base size: 0 by 0
		window gravity: NorthWest
WM_PROTOCOLS(ATOM): protocols  WM_DELETE_WINDOW, WM_TAKE_FOCUS, _NET_WM_PING, _NET_WM_SYNC_REQUEST
WM_CLASS(STRING) = "Eclipse", "Eclipse"
WM_ICON_NAME(STRING) = "eclipse-workspace - Eclipse IDE "
_NET_WM_ICON_NAME(UTF8_STRING) = "eclipse-workspace - Eclipse IDE "
WM_NAME(STRING) = "eclipse-workspace - Eclipse IDE "
_NET_WM_NAME(UTF8_STRING) = "eclipse-workspace - Eclipse IDE "

I startted this discussion because I am not sure if I need to fill an issue.
I am not too familiar with the code base, so all my attempts to understand why it is not working and fix it failed.
Can someone help me to fix it.

**Comment by VuiMuich** [comment]
This is related to #880, so no need to open a new issue.
For some reason Java styled applications have a tendency to do this.

**Reply by Fay3V** [reply]
Thanks for your response. Do you know anything I can try locally?
